### PR TITLE
Kansas Additional Exemption for Disabled Veterans 2023

### LINKED ISSUE #4969: Kansas Additional Exemption for Disabled Veterans 2023
Disabled veterans with a 100% disability rating and were honorably discharged can receive an additional exemption of $2,250.

[Reference](https://www.ksrevenue.gov/pdf/ip23.pdf)

### PR DESCRIPTION
**Pull request by hua7450**
Fixes #4969

### CODE REVIEW

File: policyengine_us/parameters/gov/states/ks/tax/income/exemptions/veterans_matched.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ks/tax/income/exemptions/veterans_matched.yaml:1]: Lets rename to `in_effect.yaml`

File: policyengine_us/tests/policy/baseline/gov/states/ks/tax/income/exemptions/ks_exemptions.yaml [hunk 1]
```diff
@@ -13,3 +13,12 @@
     state_code: KS
   output:
     ks_exemptions: 2_250 * 2
+
+- name: Two exemptions in 2022
+  period: 2023
+  input:
+    ks_disabled_veteran
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ks/tax/income/exemptions/ks_exemptions.yaml:20]: need value

File: policyengine_us/variables/gov/states/ks/tax/income/exemptions/disabled_veteran/ks_disabled_veteran_exemptions.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    entity = TaxUnit
+    label = "Kansas disabled veteran exemptions"
+    unit = USD
+    definition_period = YEAR
+    reference = "https://law.justia.com/codes/kansas/chapter-79/article-32/section-79-32-121/"
+    defined_for = StateCode.KS
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.ks.tax.income.exemptions
+        return p.veterans_matched * adds(tax_unit, period, ["ks_disabled_veteran_exemptions_person"])
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ks/tax/income/exemptions/disabled_veteran/ks_disabled_veteran_exemptions.py:15]: separate into two lines

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/ks/tax/income/exemptions/disabled_veteran/ks_disabled_veteran_exemptions.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ks/tax/income/exemptions/disabled_veteran/ks_disabled_veteran_exemptions.py:1]: lets just make one file:
`ks_disabled_veterans_exemption_eligible_person` which checks for veteran status and disability - and then add it together in the main exemption file

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/ks/tax/income/exemptions/amount/disabled_veteran.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+metadata:
+  period: year
+  unit: currency-USD
+  label: Kansas disabled veteran exemption
+  reference:
+  - title: 2023 Form K-40 instructions
+    href: https://www.ksrevenue.gov/pdf/ip23.pdf#page=6
+  - title: 2023 Form K-40 
+    href: https://www.ksrevenue.gov/pdf/k-4023.pdf#page=1
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ks/tax/income/exemptions/amount/disabled_veteran.yaml:13]:
```suggestion
    href: https://www.ksrevenue.gov/pdf/k-4023.pdf#page=2
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/ks/tax/income/exemptions/ks_disabled_veteran_exemptions_person.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+from policyengine_us.model_api import *
+
+
+class ks_disabled_veteran_exemptions_eligible_person(Variable):
+    value_type = float
+    entity = Person
+    label = "Kansas disabled veteran exemptions for each person"
+    unit = USD
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ks/tax/income/exemptions/ks_disabled_veteran_exemptions_person.py:8]:
```suggestion
    value_type = bool
    entity = Person
    label = "Eligible person for the Kansas disabled veteran exemption"
```

File: policyengine_us/parameters/gov/states/ks/tax/income/exemptions/in_effect.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ks/tax/income/exemptions/in_effect.yaml:1]: My mistake here, 
we should have
`base.yaml` - base exemptions amount 
disabled_veteran folder with files:
`base.yaml`
`in_effect.yaml`

File: policyengine_us/parameters/gov/states/ks/tax/income/exemptions/amount/base.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+description: Kansas provides this exemption amount for each person in a filing unit.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ks/tax/income/exemptions/amount/base.yaml:1]:
```suggestion
description: Kansas provides the following exemption amount for each person in a filing unit.
```

File: policyengine_us/parameters/gov/states/ks/tax/income/exemptions/amount/disabled_veteran.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+description: Kansas provides this exemption amount for disabled veterans.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ks/tax/income/exemptions/amount/disabled_veteran.yaml:1]:
```suggestion
description: Kansas provides the following exemption amount for disabled veterans.
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/household/demographic/person/is_veteran.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/household/demographic/person/is_veteran.py:1]: @MaxGhenis 
we have been using `veterans_benefits > 0` for the determination of whether the person is a veteran or not, I think adding a bool variable would be appropriate given how we handle other situations like parent etc. 
thoughts? 

@hua7450 if confirmed
could we add a formula to check if `veterans_benefits > 0` and file an issue to consolidate in the rest of the repo

File: policyengine_us/parameters/gov/states/ks/tax/income/exemptions/disabled_veteran/base.yaml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+description: Kansas provides the following exemption amount for disabled veterans.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ks/tax/income/exemptions/disabled_veteran/base.yaml:1]:
```suggestion
description: Kansas provides the following exemption amount for each disabled veteran.
```

File: policyengine_us/variables/gov/states/ks/tax/income/exemptions/ks_disabled_veteran_exemptions_person.py [hunk 8]
```diff
@@ -0,0 +1,20 @@
+from policyengine_us.model_api import *
+
+
+class ks_disabled_veteran_exemptions_eligible_person(Variable):
+    value_type = bool
+    entity = Person
+    label = "Eligible person for the Kansas disabled veteran exemption"
+    unit = USD
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ks/tax/income/exemptions/ks_disabled_veteran_exemptions_person.py:8]:
```suggestion
```

**Review by PavelMakarchuk — APPROVED** [review]
Minor format

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**